src/Mobs/AI/BehaviorTree.cpp

```diff
@@ -0,0 +1,577 @@
+#include "Globals.h"
+
+#include "BehaviorTree.h"
+#include "Blackboard.h"
+
+#include "../Monster.h"
+
+#include "../../WorldStorage/NamespaceSerializer.h"
+#include "../../World.h"
+
+
+// This logging is far too noisy to always be on
+#ifdef ABC
+
+	template <typename... Args>
+	static void BehaviorLog(cBlackboard & a_Blackboard, const char * a_FormatString, Args && ... a_Args)
+	{
+		auto Msg = fmt::format(a_FormatString, std::forward<Args>(a_Args)...);
+		FLOG("{0}: {1}", a_Blackboard.Self().GetUniqueID(), Msg);
+	}
+
+	#define BT_LOG(...) BehaviorLog(a_Blackboard, __VA_ARGS__)
+#else
+	#define BT_LOG(...)
+#endif
+
+
+
+namespace BehaviorTree
+{
+
+const char * to_string(eStatus a_Status)
+{
+	switch (a_Status)
+	{
+		case eStatus::bsUntouched: return "Untouched";
+		case eStatus::bsSuccess:   return "Success";
+		case eStatus::bsFailure:   return "Failure";
+		case eStatus::bsRunning:   return "Running";
+		case eStatus::bsCancelled: return "Cancelled";
+	}
+	UNREACHABLE("Invalid BehaviorTree::eStatus");
+}
+
+
+
+
+
+std::unique_ptr<cTask> CreateBehaviourTree(eMonsterType a_MobType, const AString & a_CustomFileName)
```

Comment:
Maybe have a separate constructor - one from string (for custom path) and one from mob type (for default)